### PR TITLE
Fix parsing of attribute 188 on seagate drives

### LINKED ISSUE #522: [BUG] Seagate Drive Command Timeout with Huge Raw Value
**Describe the bug**
I have a couple Seagate drives showing huge raw value for 188 Command Timeout, and it is marked as failed in Scrutiny. Please see screenshot below.

Seagate drives use this field's raw value to represent a combination of 3 integers (total command timeouts, commands completed between 5s and 7.5s, commands completed >7.5s). Therefore, the raw value needs to be decoded before being used to determine drive's failure.

In my case, the raw value of "4295032833" represents 1 command timeout, 1 command >5s and <7.5s, and 1 command >7.5s. This does not cross the threshold to be considered as failure.

Please see related answer at https://superuser.com/a/1747851 and [Seagate SMART Attribute Spec documentation](https://t1.daumcdn.net/brunch/service/user/axm/file/zRYOdwPu3OMoKYmBOby1fEEQEbU.pdf).

**Expected behavior**
Raw value to be decoded before being used to determine drive's failure.

**Screenshots**
image
image

### PR DESCRIPTION
**Pull request by kaysond**
Fixes #522

### CODE REVIEW

**Comment by AnalogJ** [comment]
this looks fantastic, thanks @kaysond

File: webapp/backend/pkg/thresholds/ata_attribute_metadata.go [hunk 1]
```diff
@@ -662,13 +667,33 @@ var AtaMetadata = map[int]AtaAttributeMetadata{
...
+			if len(pieces) == 3 {
+				int_pieces := make([]int, len(pieces))
+				var err error
+				for i, s := range pieces {
+					int_pieces[i], err = strconv.Atoi(s)
+					if err != nil {
+						return rawValue
+					}
+				}
+				if int_pieces[2] >= int_pieces[1] && int_pieces[1] >= int_pieces[0] {
```
Comment (firasdib) [inline webapp/backend/pkg/thresholds/ata_attribute_metadata.go:688]: This logic is slightly flawed. The first number indicates total command timeouts, the second indicates all commands that took over 5 seconds to complete, and the third all commands that took over 7.5 seconds to complete.

See:
```
Raw Usage

Raw [1 - 0] = Total # of command timeouts, with Max hold of FFFFh

Raw [3 - 2] = Total # of commands with > 5 second completion, including those > 7.5 seconds

Raw [5 - 4] = Total # of commands with > 7.5 second completion
```

**Comment by BryanPinsker** [comment]
In v0.7.2 I am still seeing the same failure on my seagate exos: 
image

While, smart continues to report OK.

**Comment by firasdib** [comment]
Does this assume that you use something along the lines of this?

```
 - device: /dev/sde
   type: 'sat'
   commands:
      metrics_smart_args: '-xv 188,raw16 --xall --json -T permissive'
```

**Comment by BryanPinsker** [comment]
Using the omnibus image, and it does not seem to support it.

**Comment by BryanPinsker** [comment]
This was my error, the issue is resolved for me on the latest beta version.

File: webapp/backend/pkg/thresholds/ata_attribute_metadata.go (hunk 1 above)
Comment (kaysond) [reply webapp/backend/pkg/thresholds/ata_attribute_metadata.go:688]: It's actually correct. When the string gets split, the indices are backwards from the byte order, and the int pieces variable uses that same backwards ordering

**Comment by kaysond** [comment]
> Does this assume that you use something along the lines of this?
> 
> ```
>  - device: /dev/sde
>    type: 'sat'
>    commands:
>       metrics_smart_args: '-xv 188,raw16 --xall --json -T permissive'
> ```

@firasdib depends. Smartctl automatically applies those arguments for seagate drives based on a few patterns. If for some reason your drive doesn't match, you'll have to add the arguments. In my case, I didn't need them

File: webapp/backend/pkg/thresholds/ata_attribute_metadata.go (hunk 1 above)
Comment (firasdib) [reply webapp/backend/pkg/thresholds/ata_attribute_metadata.go:688]: @kaysond Thank you for the clarification!

**Comment by lexxxel** [comment]
@firasdib and @kaysond thanks for the post, my `ST8000VN004-2M2101` did not fall into the automatic pattern. With the manual change, everything worked as expected. 

PS: I see that discussion went on in the bug thread, sorry.

**Comment by honmashinsei** [comment]
still here, see #655 
I'm having issues with my Seagate Ironwolf Pro 16TB (ST16000NE000-2RW103).
Other Seagate drives are fine (so far)

**Pull request merged**